libs/sdk-ui-dashboard/src/model/types/commonTypes.ts

```diff
@@ -185,6 +192,31 @@ export type DashboardContext = {
     dataProductId?: string;
 };
 
+/**
+ * @alpha
+ */
+export type DashboardTransformFn = (
+    dashboard: IDashboard<ExtendedDashboardWidget>,
+) => IDashboard<ExtendedDashboardWidget> | undefined;
+
+/**
+ * @alpha
+ */
+export type DashboardModelCustomizationFns = {
+    /**
+     * Optionally provide a function that will be used during dashboard initialization of an existing dashboard.
+     * This function will be called after the dashboard is loaded from backend and before it is dispatched for
+     * cleanup, sanitization and storage in the Dashboard component state.
+     *
+     * @remarks
+     * -  If the function is not defined, results in an error or returns `undefined`, then the original
+     *    dashboard will be used as-is.
+     * -  To test whether the dashboard object is new or existing, check for the presence of the `ref`
```

Comment:
Should we add some typeguard for this?

Reply:
actually.. the remark about new/existing dashboard was invalid.k the function will be called only for existing dashboards

initially was done so that it can be new/existing but then eventually realized its too much hassle so demoted to just existing. once we want to customize initial empty dashboard, we will add new function here. so the type guard will be unnecessary and things will be divided at the top.

Reply:
fixed comment